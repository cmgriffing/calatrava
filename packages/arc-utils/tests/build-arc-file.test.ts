import * as fs from "fs-extra";
import * as child_process from "child_process";

test("Build Arc file from example", () => {
  child_process.execSync(
    "npx ts-node scripts/build-arc-file.ts config.test.json"
  );

  const arcContents = fs.readFileSync("./app.arc", {
    encoding: "utf8",
  });

  expect(arcContents).toMatchSnapshot();
});
