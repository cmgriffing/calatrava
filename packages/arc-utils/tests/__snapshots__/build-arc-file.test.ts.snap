// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Build Arc file from example 1`] = `
"@app
example-calatrava-project

@http

/login
  method post
  src src/http/auth/login

/register
  method post
  src src/http/auth/register

/verify
  method post
  src src/http/auth/verify

/*
  method options
  src src/http/options




@tables

core
  partitionKey *String
  sortKey **String


@tables-indexes

core
  sortKey *String
  partitionKey **String
  name sortKey

core
  tertiaryKey *String
  partitionKey **String
  name tertiaryKey


@aws

@aws
timeout 10
# profile default
# region us-west-1

"
`;
